Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp

```diff
@@ -40,77 +39,86 @@ CurvilinearParameters make_params() {
   return {cov, Vector3D(0, 0, 1), Vector3D(100, 1000, 400), -1, 0};
 }
 
-TrackState<SourceLink, CurvilinearParameters> make_rand_trackstate() {
-  return {make_params()};
-}
-
 using ParVec_t = BoundParameters::ParVector_t;
 using CovMat_t = BoundParameters::CovMatrix_t;
 
+struct ParameterContainer {
+  SourceLink sl;
+  std::optional<Measurement<SourceLink, eLOC_0, eLOC_1, eQOP>> meas3d;
+  std::optional<Measurement<SourceLink, eLOC_0, eLOC_1>> meas2d;
+  std::optional<BoundParameters> predicted;
+  std::optional<BoundParameters> filtered;
+  std::optional<BoundParameters> smoothed;
+  CovMat_t jacobian;
+  double chi2;
+  double pathLength;
+};
+
 // std::pair<TrackState<SourceLink, BoundParameters>,
 // std::unique_ptr<FittableMeasurement<SourceLink>>>
-auto make_trackstate(size_t dim = 3) {
+template <typename track_state_t>
+auto fill_trackstate(track_state_t& ts, TrackStatePropMask mask,
```

Comment:
```suggestion
auto makeTrackState(track_state_t& ts, TrackStatePropMask mask,
```
See naming guidelines. Could you add a quick comment that this also modifies the input track state.

Reply:
He **just** changed it from `fill` to `make` after I asked.

Reply:
@paulgessinger was actually right about the `fill...`, will become `fillTrackState`

Reply:
Ah, Github isn't helpful when too many things are written in a short time^^

Reply:
done

Reply:
oh, forgot the comment - will add it soon

Reply:
done